### ISSUE TITLE
Zoom sensitivity

### ISSUE THREAD

**Issue by giannis196** [godotengine/godot-proposals#579]
Describe the project you are working on: I am working on a simple idle game project

Describe the problem or limitation you are having in your project: the zoom sensitivity is so sensitive that only 1 role step is enough to go too far away 

Describe the feature / enhancement and how it helps to overcome the problem or limitation: if we can set the zoom sensitivity then we will not get out of our scene every time we roll the mouse or it will not take us like 5 minutes to zoom to the place we want

and so...
I want to see in the next versions of godot a zoom sensitivity setting

Right now there is no zoom sensitivity, there is an option only for moving the cameras orbit sensitivity and it's called "orbit sensitivity" (Editor settings/General/3D/Navigation feel) but there is no option for zoom sensitivity... and changing the orbit sensitivity the zoom sensitivity is not affected.

Zoom inertia is not zoom sensitivity it only tells how fast should zoom behave so it's not affecting the zoom step size

So, if possible to put a zoom sensitivity in the next version of godot that would be fine

Thank you!

**Comment by groud** [comment]
Please fill in the issue template, it is mandatory.

**Comment by giannis196** [comment]
> 
> 
> Please fill in the issue template, it is mandatory. I will reopen once this is done.

ok, i fill the issue template (the applicable lines)

**Comment by groud** [comment]
Thanks. I'll edit it so it's more readable.

**Comment by Calinou** [comment]
If 1 zoom step is too much, then you may be working at the wrong scale. In 3D, it's recommended to follow "1 unit = 1 meter" (roughly).

**Comment by xeonmc** [comment]
zoom is a scaling operation so it should be logarithmic. Why not make each increment multiply/divide the focal length by a semitone inteval?

**Comment by Calinou** [comment]
> zoom is a scaling operation so it should be logarithmic not additive. Why not make each increment multiply/divide the focal length by a semitone inteval (12 steps for an octave)?

As far as I understand, zoom is already logarithmic in both the 2D and 3D editors.

**Comment by xeonmc** [comment]
> 
> 
> > zoom is a scaling operation so it should be logarithmic not additive. Why not make each increment multiply/divide the focal length by a semitone inteval (12 steps for an octave)?
> 
> As far as I understand, zooming is already logarithmic in both the 2D and 3D editors.

Pardon my ignorance, I've just discovered Godot and is still exploring. Does the 3D editor's zoom operation operate on the viewport's projection matrix focal length, or the distance-to-object?

**Comment by Calinou** [comment]
> Does the 3D editor's zoom operation operate on the viewport's projection matrix focal length, the abstracted FOV angle, or the distance-to-object?

It operates on the distance to the zoom origin, which is in units. This is the distance to the object if the zoom origin is at the same location as an object. It just does so in a non-linear fashion. You can view how many units you are from the zoom origin by looking at the left while using the mouse wheel. The distance to the zoom origin in units will be displayed below the meter that appears temporarily.

**Comment by xeonmc** [comment]
> 
> 
> > Does the 3D editor's zoom operation operate on the viewport's projection matrix focal length, the abstracted FOV angle, or the distance-to-object?
> 
> It operates on the distance to the zoom origin, which is in units. This is the distance to the object if the zoom origin is at the same location as an object. It just does so in a non-linear fashion. You can view how many units you are from the zoom origin by looking at the left while using the mouse wheel. The distance to the zoom origin in units will be displayed below the meter that appears temporarily.

Curious, then, that the OP would experience a scale-dependent problem like this, if the operation is already scale-free/multiplicative. The only situation that I can think of where you would have such an issue is if your zoom origin is very far from the scene.

**Comment by lowagner** [comment]
this would be nice to have when working on a laptop away from a mouse (zoom is too insensitive), and i'd be willing to add it.  doesn't look like this is accepted yet, though.

**Comment by actforjason** [comment]
Maybe this will be helpful https://github.com/godotengine/godot-proposals/issues/1470#issuecomment-688487903
